setup.py

```diff
@@ -83,6 +80,11 @@ def read(fname):
 
 
 def run_setup(with_binary):
+    if with_binary:
+        extra_opts = {}
+        extra_opts['ext_modules'] = ext_modules
```

Comment:
nit: L.84-85 can be one line and it would be simpler. Or, L.84 and L.87 are same, so it can be moved outside of the if clause.